ckan/views/dataset.py

```diff
@@ -1097,9 +1098,22 @@ def get(self, package_type: str, id: str) -> str:
         )
 
 
-def activity(package_type: str, id: str) -> Union[Response, str]:  # noqa
+def activity(
+    package_type: str,
+    id: str,
+) -> Union[Response, str]:  # noqa
+
     """Render this package's public activity stream page.
     """
+    after = h.get_request_param(u'after')
+    before = h.get_request_param(u'before')
+    activity_type = h.get_request_param(u'activity_type')
+
+    if after and before:
+        raise ValidationError(
+            {'after': ['Cannot be used together with `before']}
+        )
+
```

Comment:
These checks should be done via schema validators, not directly in the blueprints (otherwise they are not applied when using the API)

Reply:
A new validator was created [here](https://github.com/avdata99/ckan/commit/42975b649e9370a5827c7aa148223991fd588281). 
Now we allow after and before together.